src/imputors/drop.jl

```diff
@@ -1,73 +1,143 @@
-"""
-function impute!(imp::Drop, ctx::Context, data::AbstractMatrix)
-    ctx.num = size(data, 1)
-    mask = map(i -> !ismissing(ctx, data[i, :]), 1:size(data, 1))
-    return data[mask, :]
+        return table
+    end
+end
+
+
+struct DropVars <: Imputor
+    vardim::Int
+    context::AbstractContext
 end
 
 """
-    impute!(imp::Drop, ctx::Context, data::DataFrame)
+    DropVars(; vardim=2, context=Context())
+
 
-Finds the missing rows in the `DataFrame` and deletes them.
+Finds variables with too many missing values in a `AbstractMatrix` or `Tables.table` and
+removes them from the input data.
 
-NOTE: this isn't quite as fast as `dropnull` in `DataFrames`s as we're using an arbitrary
-`missing` function rather than using the precomputed `dt.isnull` vector of bools.
+# Keyword Arguments
+* `vardim=2::Int`: Specify the dimension for variables in matrix input data
+* `context::AbstractContext`: A context which keeps track of missing data
+  summary information
 
-# Arguments
-* `imp::Drop`: this `Imputor` method
-* `ctx::Context`: contextual information for missing data
-* `data::DataFrame`: the data to impute
+# Examples
+```jldoctest
+julia> using Impute: DropVars, Context, impute
 
-# Returns
-* `DataFrame`: our data with the missing rows removed.
+julia> M = [1.0 2.0 missing missing 5.0; 1.1 2.2 3.3 missing 5.5]
+2×5 Array{Union{Missing, Float64},2}:
+ 1.0  2.0   missing  missing  5.0
+ 1.1  2.2  3.3       missing  5.5
+
+julia> impute(DropVars(; vardim=1, context=Context(; limit=0.2)), M)
+1×5 Array{Union{Missing, Float64},2}:
+ 1.1  2.2  3.3  missing  5.5
+```
 """
-function impute!(imp::Drop, ctx::Context, data::DataFrame)
-    ctx.num = size(data, 1)
-    m = typeof(data).name.module
-    m.deleterows!(data, findall(r -> ismissing(ctx, r), m.eachrow(data)))
-    return data
+DropVars(; vardim=2, context=Context()) = DropVars(vardim, context)
+
+function impute!(imp::DropVars, data::AbstractMatrix)
+    return filtervars(imp, data) do var
+        try
+            imp.context() do c
+                for x in var
+                    ismissing(c, x)
+                end
+            end
+            return true
+        catch e
+            if isa(e, ImputeError)
+                return false
+            else
+                rethrow(e)
+            end
+        end
+    end
+end
+
+function impute!(imp::DropVars, table)
+    @assert istable(table)
+    cols = Tables.columns(table)
+
+    cnames = Iterators.filter(propertynames(cols)) do cname
+        try
+            imp.context() do c
+                col = getproperty(cols, cname)
+                for i in 1:length(col)
+                    ismissing(c, col[i])
+                end
+            end
+            return true
+        catch e
+            if isa(e, ImputeError)
+                return false
+            else
+                rethrow(e)
+            end
+        end
+    end
+
+    table = Tables.select(table, cnames...) |> materializer(table)
+    return table
```

Comment:
please use `return materializer(table)`

Reply:
See again, I agree with Nick but will not block the PR over it